intlogger.go

```diff
@@ -353,32 +375,69 @@ func (l *intLogger) logPlain(t time.Time, name string, level Level, msg string,
 				}
 			}
 
-			var key string
-
+			// Convert the field key to a string.
 			switch st := args[i].(type) {
 			case string:
 				key = st
 			default:
 				key = fmt.Sprintf("%s", st)
 			}
 
-			if strings.Contains(val, "\n") {
-				l.writer.WriteString("\n  ")
-				l.writer.WriteString(key)
-				l.writer.WriteString("=\n")
-				writeIndent(l.writer, val, "  | ")
-				l.writer.WriteString("  ")
-			} else if !raw && needsQuoting(val) {
+			// Optionally apply the ANSI "dim" (faint) and "bold"
+			// SGR values to the key.
+			if l.fieldColor != ColorOff {
+				color := color.New(color.Faint, color.Bold)
+				key = color.Sprint(key)
+			}
+
+			// Values may contain multiple lines, and that format
+			// is preserved, with each line prefixed with a "  | "
+			// to show it's part of a collection of lines.
+			//
+			// Values may also neeq quoting, if not all the runes
+			// in the value string are "normal", like if they
+			// contain ANSI escape sequences.
+			switch {
+			case strings.Contains(val, "\n"):
+				key = "\n  " + key
+
+				valStrBuilder := &strings.Builder{}
+				valStrWriter := newWriter(valStrBuilder, ColorOff)
+
+				valStrBuilder.WriteString("\n")
+
+				if l.fieldColor != ColorOff {
+					color := color.New(color.Faint)
+					writeIndent(valStrWriter, val, color.Sprint("  | "))
+				} else {
+					writeIndent(valStrWriter, val, "  | ")
+				}
+				valStrWriter.Flush(level)
+
+				valStrBuilder.WriteString("  ")
+
+				val = valStrBuilder.String()
+			case !raw && needsQuoting(val):
+				val = strconv.Quote(val)
+			}
+
+			// If the key contains a newline, because the value itself
+			// contains newlines, then we pad with an extra space.
+			if !strings.Contains(key, "\n") {
 				l.writer.WriteByte(' ')
-				l.writer.WriteString(key)
-				l.writer.WriteByte('=')
-				l.writer.WriteString(strconv.Quote(val))
+			}
+			l.writer.WriteString(key)
+
+			if l.fieldColor != ColorOff {
+				faintColor := color.New(color.Faint)
+				faintColor.Fprint(l.writer, "=")
+				resetColor := color.New(color.Reset)
+				resetColor.Fprint(l.writer, "")
```

Comment:
I'm confused why we're checking for color again when we did color up in the other branch?

Reply:
When `ColorHeaderAndFields` is used, the `=` part of `key=val` is faintly colored, but not bold.

### With
<img width="558" alt="Screen Shot 2022-08-29 at 1 43 18 PM" src="https://user-images.githubusercontent.com/14850816/187264586-eb4a6d56-5122-42da-a96e-dc5975a882d3.png">


### Without
<img width="557" alt="Screen Shot 2022-08-29 at 1 42 46 PM" src="https://user-images.githubusercontent.com/14850816/187264495-c27d9a46-76a8-437f-819e-45e48119c170.png">

Reply:
I see, you're trying to color the = only faint, gotcha. Let's pull those `color.New`s outside and make them globals. I try really hard to keep allocations as low as possible in this code.

Reply:
Sounds good, can do 👍

Reply:
Should be addressed with https://github.com/hashicorp/go-hclog/pull/118/commits/7c2f5c4ed6b2d2235b579b92b8a4a43acf364208